### DISCUSSION TITLE
Battery control

### DISCUSSION THREAD

**Discussion by sedlarm** [wills106/homeassistant-solax-modbus#883] [Q&A]
Hello guys, 
thanks for oustanding work with this integration!

I need help with achieving simple scenario - don't allow to charge battery in some time window (based on current electricity price). So for example, if current day PV forecast is good, I would like to export energy to grid in the morning a charge the battery later - after noon, where export prices are at low level.

I read the documentation and try to build automation based on setting Power Control to "Battery Control" and Active Power = 0 (no charge/discharge battery). But after setting those values, I see, that battery is discharging also - going to grid and house load. Could you recommend me the setting to achieve taks to no charge and discharge battery during some hours (using Autorepeat duration = 3600) using Modbus registers? All other parts of my automation -  PV prediction and current price is working very well, but don't understand what setup on Solax Modbus integration.

Thanks a lot!

**Comment by Tangiers2020** [comment]
Hey, did you ever get this working, I'm in a similar boat :-( but cant find the answer.

**Comment by cschlipf** [comment]
Which type of inverter?

**Comment by PatrikTrestik** [comment]
There is new (or newly found) documentation for Modbus Remote control: [SolaX_VPP function Definition of ESS](https://kb.solaxpower.com/solution/detail/2c9fa4148ecd09eb018edf67a87b01d2)

From documentation it is obvious that SolaX developers changed something or we do not get clear view how it should work.
This new description shows some light. I'm testing it and will prepare PR once I have working version.

**Reply by johny-mnemonic** [reply]
I have seen your PR for this, but I honestly do not understand what it changes.
I also do not see how to change the `ModbusPowerControl` mode. 
The integration only have a sensor for it (`sensor.solax_modbus_power_control`) which changes between `Disabled` and `Enabled Power Control` when you set the `select.solax_remotecontrol_power_control` to some of the artificial modes supported by the integration.

Would you advice how to for example use the `Enabled SOC Target Control` or other modes you have now added?

I am using `number.solax_battery_charge_max_current` to block battery charge, but it has couple disadvantages. First it's writing to EEPROM and today I have also discovered, that once it is set to 0 and some of the power control modes is active, it completely blocks PV production😱 Not sure if this is some new bug, or whether I just didn't find it earlier, but I would like to use modbus control to stop battery charge instead of EEPROM value anyway and some of those modes looks like they could be used for that.

**Reply by PatrikTrestik** [reply]
> I have seen your PR for this, but I honestly do not understand what it changes. I also do not see how to change the `ModbusPowerControl` mode. The integration only have a sensor for it (`sensor.solax_modbus_power_control`) which changes between `Disabled` and `Enabled Power Control` when you set the `select.solax_remotecontrol_power_control` to some of the artificial modes supported by the integration.
> 
> Would you advice how to for example use the `Enabled SOC Target Control` or other modes you have now added?
> 
> I am using `number.solax_battery_charge_max_current` to block battery charge, but it has couple disadvantages. First it's writing to EEPROM and today I have also discovered, that once it is set to 0 and some of the power control modes is active, it completely blocks PV production😱 Not sure if this is some new bug, or whether I just didn't find it earlier, but I would like to use modbus control to stop battery charge instead of EEPROM value anyway and some of those modes looks like they could be used for that.

At first, my PR is only adding two new values for Modbus Remote control. Not changing anything else.
I'll post full PR only when I fully understand how it works.

Current implementation of Modbus Remote Control is based on @wills106 geus work. There was no documentation available at all.
Newly found document slightly contradicts what is now implemented.
But it requires testing before I'll be sure what and how to change. I have implementation in Node Red and it works promissingly well.

I promisse that during January I'll send PR with working code.

**Reply by johny-mnemonic** [reply]
OK. Got it.
Good luck than and thanks for your work on this!
